tests/test-org-ql.el

```diff
@@ -2111,7 +2111,84 @@ with keyword arg NOW in PLIST."
           (it "Refuses to link to non-file-backed buffer"
             (expect (var-after-link-save-open 'org-ql-view-buffers-files link-buffer query
                                               :buffer link-buffer)
-                    :to-throw 'user-error '("Views that search non-file-backed buffers can’t be linked to"))))))
+                    :to-throw 'user-error '("Views that search non-file-backed buffers can’t be linked to"))))
+        (describe "while completion for files/buffers"
+          (describe "contracting org-ql-view-buffers-files"
+            (it "list of files to \"org-agenda-files\""
+              (spy-on 'org-agenda-files :and-return-value temp-filenames)
+              (expect (org-ql-view--contract-buffers-files temp-filenames) :to-equal "org-agenda-files"))
+            (it "list of files to \"org-directory\""
+              (spy-on 'org-agenda-files :and-return-value '())
+              ;; Also indirectly tests org-ql-search-directories-files
+              (let ((org-directory temp-dir))  ;; the :var binding does not work? https://github.com/jorgenschaefer/emacs-buttercup/issues/127
+                (expect (org-ql-view--contract-buffers-files temp-filenames) :to-equal "org-directory")))
+            (it "to \"buffer\" when passing current-buffer"
+              (with-current-buffer (org-ql-test-data-buffer "data.org")
+                (expect (org-ql-view--contract-buffers-files (current-buffer)) :to-equal "buffer")))
+            (it "to \"org-agenda-files\" from symbol values ('org-agenda-files or #'org-agenda-files)"
+              (spy-on 'org-agenda-files :and-return-value temp-filenames)
+              (expect (org-ql-view--contract-buffers-files 'org-agenda-files) :to-equal "org-agenda-files")
+              (expect (org-ql-view--contract-buffers-files #'org-agenda-files) :to-equal "org-agenda-files"))
+            (it "arbitarary list of buffers/files"
+              (let ((value1 '("a.org" "b.org"))
+                    (value2 'a))
+                (expect (org-ql-view--contract-buffers-files value1) :to-equal value1)
+                ;; If the value does not result to a buffer, file, or string, throws error
```

Comment:
Nitpick: This isn't grammatical.  How about something like, "Signals error if value is not a buffer, file, or string."

(Note also the use of "signal" rather than "throw": in Lisp, `throw` is something else.)

Generally, please check the comments in this PR and ensure they're complete, grammatical sentences, ending in a period.  :)